EIPS/eip-draft_transaction_representation.md

```diff
@@ -0,0 +1,67 @@
+- If the request comes from a URI source, the wallet MUST indicate the source of the request to the user, including the URI scheme, host, and port. If the transaction originates from an embedded frame (e.g. an iframe or webview), the wallet MUST indicate this to the user and add a warning that the request may not be trustworthy.
+- If the request references a contract that doesn't have verified source code, the wallet MUST indicate this to the user and add a warning that the request may not be trustworthy. Additionally, a warning badge MAY be displayed next to any such contract.
+- If the wallet has a list of known malicious contracts, and a request references one of these contracts, the wallet MUST indicate this to the user and add a warning that the request may not be trustworthy. Additionally, a warning badge SHOULD be displayed next to any such contract.
+
+### Asset Transfer Rules
+
+- If a transfer of tokens ([ERC-20](./eip-20.md), [ERC-721](./eip-721.md), [ERC-1046](./eip-1046.md)-compatible, or [ERC-1155](./eip-1155.md)) or a chain's native currency is included in the transaction, the wallet SHOULD display the amount of tokens being transferred, the token's symbol (if available), the token's name (if available), an image of the token (if available), the token's contract address, the recipient's address, and the sender's address.
+  - If the transaction format is known, and the transaction request doesn't explicitly disallow it, the wallet SHOULD allow the user to edit the amount of asset being transferred. If/when the user does, the wallet MUST update the transaction request accordingly.
+  - If more than one transfer of the same asset is included in the transaction, the wallet MAY display the net transfer amount instead of each individual transfer.
+  - If more than one transfer of different assets is included in the transaction, the wallet SHOULD display the transfers as a trade.
+    - If the maximum slippage can be determined, the wallet SHOULD display the maximum slippage.
+    - If the determined slippage is greater than some user-defined thresholds (with sensible defaults e.g. 1%, 5%, 10%), the wallet SHOULD display progressively more severe warnings to the user.
+- If an appproval is included in a transaction, it must follow the same rules as a transfer, except that the approvals MUST NOT be represented as a trade.
+
+### Staking
+
+- If the transaction stakes native currency for validator rewards, the wallet SHOULD display the amount of native currency being staked (determined by the transaction value), the public key, and withdrawal credentials (fetched from the `DepositEvent` event).
+
+## Rationale
+
+An automatically-generated representation of a transaction is much more user-friendly and easier to understand than the raw JSON/YAML representation. Additionally, the wallet and user don't need to trust the website or contract to generate a correct representation.
```

Comment:
This belongs in motivation, I think.

Reply:
> This belongs in motivation, I think.

Reply:
This is the Rationale for why the transaction representation is automatically generated. The inspiration for this EIP was a suggestion for manual transaction representation.

Reply:
It explains the design decision of not letting the contracts decide how to represent the transactions.